### ISSUE TITLE
Code in AllSTDPSynapses::stdpLearning() seems redundant

### ISSUE THREAD

**Issue by stiber** [UWB-Biocomputing/Graphitti#200]
In particular:
```C++
   if (delta < -STDPgap) {
      // depression
      
      dw = synapticWeightModification(iEdg, W, delta);// normalize
   } else if (delta > STDPgap) {
      // potentiation
      dw = synapticWeightModification(iEdg, W, delta); // normalize
   }
   else {
      
      return;
   }
```

**Comment by stiber** [comment]
For that matter, I don't think `modDelta` is used; should consult with @sky1204 on this. Same goes for `AllSTDPSynapses::synapticWeightModification()`

**Comment by sky1204** [comment]
I included it because some STDP implementations use modDelta. I did not get to use those but if someone wants to implement them in the future, it will be easier. We can remove it if we want. synapticWeightModification() is to make the implementation of different STDP functions easier. I did not use it because when I made a separate function, I had already done some of the analysis and I wanted to maintain consistency (not that using the function would have changes something). This function I would suggest keeping and replacing the code for updating weights with the function.
